.github/workflows/test.yml

```diff
@@ -11,152 +11,119 @@ on:
+              
     strategy:
+      fail-fast: false
       matrix:
-        rust-version: [stable, beta, nightly]
+        config:
+          - {os: windows-latest, r: 'release', rust-version: 'stable-x86_64-pc-windows-msvc'}
+          - {os: windows-latest, r: 'release', rust-version: 'stable-i686-pc-windows-msvc'}
+          - {os: macOS-latest,   r: 'release', rust-version: 'stable'}
+          - {os: ubuntu-20.04,   r: 'release', rust-version: 'stable', rspm: "https://packagemanager.rstudio.com/cran/__linux__/focal/latest"}
+          - {os: ubuntu-20.04,   r: 'release', rust-version: 'nightly', rspm: "https://packagemanager.rstudio.com/cran/__linux__/focal/latest"}
+          # R-devel fails on Ubuntu
+          # - {os: ubuntu-20.04,   r: 'devel',   rust-version: 'nightly', rspm: "https://packagemanager.rstudio.com/cran/__linux__/focal/latest"}   
 
-    steps:
-      - uses: actions/checkout@v2
-      - uses: actions-rs/toolchain@v1
-        with:
-          toolchain: ${{ matrix.rust-version }}
-      - name: Build
-        id: build
-        run: cargo build -vv --features use-bindgen
-        env:
-          LIBRSYS_BINDINGS_OUTPUT_PATH: generated_bindings
-      - name: Run tests
-        run: cargo test -vv --features use-bindgen -- --nocapture --test-threads=1
-      - name: Upload generated bindings
-        if: steps.build.outcome == 'success'
-        uses: actions/upload-artifact@main
-        with:
-          name: Linux ${{ matrix.rust-version }} generated bindings
-          path: generated_bindings
-      - name: Run tests on precomputed bindings shipped with libR-sys
-        run: cargo test -vv -- --nocapture --test-threads=1
 
+    env:
+      RSPM: ${{ matrix.config.rspm }}
 
-  test_macos:
-    name: Run tests (macOS)
-    runs-on: macos-latest
-    strategy:
-      matrix:
-        rust-version: [stable]
+    # PowerShell core is available on all platforms and can be used to unify scripts
+    defaults:
+      run:
+        shell: pwsh
     steps:
       - uses: actions/checkout@v2
-      - uses: actions-rs/toolchain@v1
+      
+      
+      - name: Set up `R`
+        uses: r-lib/actions/setup-r@v1
         with:
-          toolchain: ${{ matrix.rust-version }}
-      - name: Install dependencies
-        run: brew install llvm r
-      - name: Set dynamic environment variables
-        run: echo "LIBCLANG_PATH=$(brew --prefix llvm)/lib" >> $GITHUB_ENV
-      - name: Build
-        id: build
-        run: cargo build -vv --features use-bindgen
-        env:
-          LIBRSYS_BINDINGS_OUTPUT_PATH: generated_bindings
-      - name: Run tests
-        run: cargo test -vv --features use-bindgen -- --nocapture --test-threads=1
-      - name: Upload generated bindings shipped with libR-sys
-        if: steps.build.outcome == 'success'
-        uses: actions/upload-artifact@main
+          r-version: ${{ matrix.config.r }}
+
+      - name: Set up `rust`
```

Comment:
I think this should be `Rust` instead of `` `rust` ``. In general, I think names of programming languages should not be set in backticks, but more specifically here I don't think the backticks are converted into code font.